packages-exp/auth-exp/src/model/user_credential.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthCredential } from './auth_credential';
+import { User } from './user';
+
+export enum OperationType {
+  LINK = 'link',
+  REAUTHENTICATE = 'reauthenticate',
+  SIGN_IN = 'signIn'
+}
+
+export interface UserCredential {
```

Comment:
should this go in external types? probably

Reply:
yep